### PR TITLE
Add Assignments

### PR DESCRIPTION
**Pull request by prashantkn94**

### CODE REVIEW

File: Assignment1/TripletSum.cpp [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+	
+    sort(data.begin(), data.end());
+
+    // Iterate the vector from light to right
+    for ( int fixedIndex = 0; fixedIndex < data.size() - 2; ++fixedIndex )
+    {
+        int leftIndex = fixedIndex + 1, rightIndex = data.size() - 1;
+
+        // Sum might overflow, so use long long
+        long long int currentSUm = data[fixedIndex] + data[leftIndex] + data[rightIndex];
```
Comment (ankuriitk) [inline Assignment1/TripletSum.cpp]: you seem to be using camel casing convention of variable names. Thats fine but in that spirit currentSUm should be currentSum.

File: Assignment1/TripletSumTest.cpp [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+
+	tripletSum.setData(data);
+	tripletSum.setTargetSum(targetSum);
+
+	// Data = {1, 1, 1, 2} and targetSum = 4
+	CPPUNIT_ASSERT(tripletSum.checkTargetSum() == true);
+
+	vector<int> elements = tripletSum.getElements();
+
+	long long sum = elements[0] + elements[1] + elements[2];
```
Comment (ankuriitk) [inline Assignment1/TripletSumTest.cpp]: more tests cases to consider: negative numbers, multiple tuples matching a given sum.

File: Assignment2/InviteFriendsTest.cpp [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+    vector < pair < int, string > > friendsComingToParty = inviteFriends.getIdsWithName();
+
+    for ( int i = 0; i < friendsComingToParty.size(); ++i )
+    {
+        cout << friendsComingToParty[i].second << "\n\n";
+    }
+
+    fileName = "friends2.json";
+
+    bool failed = false;
```
Comment (ankuriitk) [inline Assignment2/InviteFriendsTest.cpp:27]: these tests are very weak - they only test readFile works..no assertions around the actual friends being output. cout at line 22 should be been replaced by assertions.

File: Assignment2/InviteFriends.cpp [hunk 4]
```diff
@@ -0,0 +1,67 @@
+#include "InviteFriends.h"
+
+using namespace std;
+
+double InviteFriends::getDistance(const double latitude, const double longitude)
```
Comment (ankuriitk) [inline Assignment2/InviteFriends.cpp]: this problem was perfect for using Object oriented programming. You could have defined a LatLng class that encapsulats distance function. Then a separate Friend class and Party class could have done rest of the work.

File: Assignment2/InviteFriendsTest.cpp [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+    if ( !inviteFriends.readFile(fileName) )
+	{
+        cout << "Cannot read the file\n\n";
+        failed = true;
+	}
+
+	CPPUNIT_ASSERT(failed == false);
+
+	friendsComingToParty = inviteFriends.getIdsWithName();
+
```
Comment (ankuriitk) [inline Assignment2/InviteFriendsTest.cpp]: tests should have included distance calculation tests separately - that should have tested negative lat/lng, north pole/south pole combo etc.

File: Assignment2/InviteFriends.cpp (hunk 4 above)
Comment (prashantkn94) [reply Assignment2/InviteFriends.cpp]: Do you mean to say that for each Friend object we store a separate LatLng object and LatLng class will calculate the distance and may contain some other info also and the Party class (currently InviteFriends) would just check and store the list of Friend objects ?

Comment (ankuriitk) [reply Assignment2/InviteFriends.cpp]: LatLng should store the lat lng, thats why it is named LatLng class.

File: Assignment2/InviteFriendsTest.cpp (hunk 3 above)
Comment (prashantkn94) [reply Assignment2/InviteFriendsTest.cpp:27]: We are not supposed to allow negative coordinates for this specific case right ?  because radius of 100 km around banglore does not include negative coordinates. Same doubt for North/South Poles.

Comment (ankuriitk) [reply Assignment2/InviteFriendsTest.cpp:27]: a well written code should be agnostic to whether it is bangalore or any other lat/lng

Comment (prashantkn94) [reply Assignment2/InviteFriendsTest.cpp:27]: Yes, I agree with you but I am a bit confused here because as far as I can see the latitude/longitude can be negative also. So, why would we avoid the negative latitude/longitude ?

Comment (ankuriitk) [reply Assignment2/InviteFriendsTest.cpp:27]: we should not avoid negative lat/lng as they are valid inputs - that is what I meant by well written code

**Comment by ankuriitk** [comment]
Anyway, I think the PR is good enough. Can we talk sometime today? Please call me at 8861302652 anytime from now till 5pm.
